src/network/tarcap/packets_handler/handlers/blocks_packet_handler.cpp

```diff
@@ -0,0 +1,57 @@
+#include "blocks_packet_handler.hpp"
+
+#include "dag/dag_block_manager.hpp"
+#include "network/tarcap/packets_handler/syncing_state.hpp"
+
+namespace taraxa::network::tarcap {
+
+BlocksPacketHandler::BlocksPacketHandler(std::shared_ptr<PeersState> peers_state,
+                                         std::shared_ptr<SyncingState> syncing_state,
+                                         std::shared_ptr<DagBlockManager> dag_blk_mgr, const addr_t &node_addr)
+    : PacketHandler(std::move(peers_state), node_addr, "BLOCKS_PH"),
+      syncing_state_(syncing_state),
+      dag_blk_mgr_(dag_blk_mgr) {}
+
+void BlocksPacketHandler::process(const PacketData &packet_data, const dev::RLP &packet_rlp) {
+  std::string received_dag_blocks_str;
+  auto it = packet_rlp.begin();
+  const bool is_final_sync_packet = (*it++).toInt<unsigned>();
+
+  for (; it != packet_rlp.end();) {
+    DagBlock block(*it++);
+    peer_->markBlockAsKnown(block.getHash());
+
+    std::vector<Transaction> new_transactions;
+    for (size_t i = 0; i < block.getTrxs().size(); i++) {
+      Transaction transaction(*it++);
+      new_transactions.push_back(transaction);
+      peer_->markTransactionAsKnown(transaction.getHash());
+    }
+
+    received_dag_blocks_str += block.getHash().toString() + " ";
+
+    auto status = syncing_state_->checkDagBlockValidation(block);
+    if (!status.first) {
+      LOG(log_wr_) << "DagBlockValidation failed " << status.second;
+      status.second.push_back(block.getHash());
+      syncing_state_->requestBlocks(packet_data.from_node_id_, status.second,
+                                    GetBlocksPacketRequestType::MissingHashes);
+      continue;
+    }
+
+    LOG(log_dg_) << "Storing block " << block.getHash().toString() << " with " << new_transactions.size()
+                 << " transactions";
+    if (block.getLevel() > peer_->dag_level_) peer_->dag_level_ = block.getLevel();
+    dag_blk_mgr_->insertBroadcastedBlockWithTransactions(block, new_transactions);
```

Comment:
when I look at it now, we are probably marking txs and block as known even if received invalid(malicious) block. Maybe we could even return bool flag from  insertBroadcastedBlockWithTransactions and mark block as well as txs as known only if true was returned...

Reply:
but is there a problem? we mark, that specific block is invalid for specific peer. I do not see a problem there

Reply:
we dont mark it as invalid but as known, but yeah maybe we can do that even if the block we received here is invalid. Not 100% sure though

Reply:
I think this is ok, invalid block will be marked for specific peer, but never broadcasted or inserted anywhere. so it does not affect anything

Reply:
yeah you are right probably. But in functions that send data to the peer and mark it as known should mark everything at the end of function I guess

Reply:
yes, that was fixed